tests/core/CoreTest.py

```diff
@@ -377,13 +377,17 @@ def setUp(self):
         self.filedict1 = {'f1': []}
 
     @staticmethod
-    def execute_run(bears):
+    def execute_run(bears, result_callback=None):
         results = []
 
         def on_result(result):
             results.append(result)
 
-        run(bears, on_result)
+        if result_callback is None:
+            result_callback = on_result
+
+        session = Session(bears, result_callback)
+        session.run()
```

Comment:
Misunderstood it on first go. Corrected. Additionally, I'm also returning `Session` object, just in case, if its needed further in code using `Session`.